### DISCUSSION TITLE
Error: Agent and Connection failed connecting to streaming RPC server

### DISCUSSION THREAD

**Discussion by adeifeadekunbi** [hoophq/hoop#624] [Q&A]
Please i am encountering the below error with agent after successfully enabling TLS for gateway

```
agent-1    | 2024-12-30T08:17:26Z       info    agent/main.go:166       version=1.31.10, platform=linux/amd64, type=dsn, mode=standard, grpc_server=gateway:8010, tls=false, tlsca=true - starting agent
agent-1    | 2024-12-30T08:17:26Z       info    agent/main.go:171       connecting to gateway:8010, tls=false   {"version": "1.31.10", "backoff": "1s"}
agent-1    | 2024-12-30T08:17:26Z       warn    agent/main.go:175       failed to connect to gateway:8010, reason=failed connecting to streaming RPC server, err=rpc error: code = Unavailable desc = connection error: desc = "error reading server preface: EOF"   {"version": "1.31.10", "backoff": "1s"}
```
Which variable should I use to pass TLSCA to the agent, or what is the correct method to enable TLS communication between the gateway and the agent for docker implementation?

Please find the attached sample docker-compose.yml file for your reference.
[dockercompose hoop.txt](https://github.com/user-attachments/files/18273091/dockercompose.hoop.txt)

**Comment by adeifeadekunbi** [comment]
I login to the gateway and i was able to setup a new agent which connected successfully but the default agent was still showing the above reported error
`agent-1    | 2025-01-03T08:21:47Z       info    agent/main.go:166       version=1.31.10, platform=linux/amd64, type=dsn, mode=standard, grpc_server= 1.2.3.01:8010, tls=true, tlsca=true - starting agent
agent-1    | 2025-01-03T08:21:47Z       info    agent/main.go:171       connecting to  1.2.3.01:8010, tls=true        {"version": "1.31.10", "backoff": "1s"}
gateway-1  | 2025-01-03T08:21:47Z       info    transport/agent.go:37   agent connected: org=c8190358-31d4-4d2d-9997-3aa41c8c5560,name=system,mode=standard,hostname=,platform=,version=,goversion=,kernel= {"connection": ""}
agent-1    | 2025-01-03T08:21:47Z       info    controller/agent.go:106 connected with success to  1.2.3.01:8010`

I then processed to create a mysql database connection but it also failing with below error
`2025-01-03T08:35:47Z       error   connections/connections.go:606  failed connecting to streaming RPC server, err=rpc error: code = Unavailable desc = connection error: desc = "error reading server preface: EOF"
`

Please can someone share details guide on how to setup hoop.dev with TLS for both agent and connections.

Thank you

**Reply by sandromello** [reply]
Hello @adeifeadekunbi 

We released a new version (1.31.15) that allows configuring the default agent to connect via TLS when the `TLS_KEY` env is set.
Update the image `hoophq/hooplabs` in your machine and configure your `.env` file with the additional environment variables:

```sh
# gateway TLS configuration
TLS_KEY=base64://<pem-base64-content>
TLS_CERT=base64://<pem-base64-content>

# default agent TLS configuration
HOOP_TLSCA=base64://<pem-base64-content>
```
